src/client.cpp

```diff
@@ -194,22 +194,25 @@ bool hasIntMember(const rapidjson::Value& value, const char* name)
     return value.HasMember(name)  &&  value[name].IsInt();
 }
 
+#define CREATE_SESSION() \
+    CurlSessionPtr sessionPtr = createSession();\
+    if (!sessionPtr)\
+        return makeError();\
+    CurlSession& session = *sessionPtr;
```

Comment:
Suggestions on improving this - doing it without macro and without code duplication or using better macro - are welcome